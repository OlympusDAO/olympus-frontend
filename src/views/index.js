export { default as Bond } from "./Bond/Bond";
export { default as ChooseBond } from "./ChooseBond/ChooseBond";
export { default as Stake } from "./Stake/Stake";
export { default as Wrap } from "./Wrap/Wrap";
export { default as Dashboard } from "./Dashboard/Dashboard";
export { default as PoolTogether } from "./33Together/33Together.jsx";
export { default as TreasuryDashboard } from "./TreasuryDashboard/TreasuryDashboard";
