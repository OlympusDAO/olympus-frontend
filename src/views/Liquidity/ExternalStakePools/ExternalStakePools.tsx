import { Check } from "@mui/icons-material";
import { Box, Typography } from "@mui/material";
import { styled } from "@mui/material/styles";
import { DataGrid, GridColDef, GridRenderCellParams, GridValueFormatterParams } from "@mui/x-data-grid";
import {
  Chip,
  OHMTokenProps,
  OHMTokenStackProps,
  SecondaryButton,
  TertiaryButton,
  TokenStack,
  Tooltip,
} from "@olympusdao/component-library";
import { useState } from "react";
import PageTitle from "src/components/PageTitle";
import { formatCurrency, formatNumber } from "src/helpers";
import { defiLlamaChainToNetwork } from "src/helpers/defiLlamaChainToNetwork";
import { normalizeSymbol } from "src/helpers/normalizeSymbol";
import { DefiLlamaPool, useGetLPStats } from "src/hooks/useGetLPStats";

const PREFIX = "ExternalStakePools";

const classes = {
  stakePoolsWrapper: `${PREFIX}-stakePoolsWrapper`,
  stakePoolHeaderText: `${PREFIX}-stakePoolHeaderText`,
  poolPair: `${PREFIX}-poolPair`,
  poolName: `${PREFIX}-poolName`,
};

const StyledPoolInfo = styled("div")(() => ({
  [`&.${classes.poolPair}`]: {
    display: "flex",
    alignItems: "center",
    justifyContent: "left",
  },

  [`& .${classes.poolName}`]: {
    marginLeft: "10px",
  },
}));

export const ExternalStakePools = () => {
  const { data: defiLlamaPools } = useGetLPStats();
  const [poolFilter, setPoolFilter] = useState("all");
  const [networkFilter, setNetworkFilter] = useState<undefined | string>(undefined);
  const stablePools =
    defiLlamaPools &&
    defiLlamaPools.filter(pool => {
      const symbols = pool.symbol.split("-");
      const stable =
        symbols.includes("DAI") ||
        symbols.includes("USDC") ||
        symbols.includes("FRAXBP") ||
        symbols.includes("OHMFRAXBP");
      return stable;
    });

  const volatilePools =
    defiLlamaPools &&
    defiLlamaPools.filter(pool => {
      const symbols = pool.symbol.split("-");
      const volatile =
        !symbols.includes("DAI") &&
        !symbols.includes("USDC") &&
        !symbols.includes("FRAXBP") &&
        !symbols.includes("OHMFRAXBP");
      return volatile;
    });

  const gOHMPools =
    defiLlamaPools &&
    defiLlamaPools.filter(pool => {
      const symbols = pool.symbol.split("-");
      const stable = symbols.includes("GOHM");
      return stable;
    });

  const poolList =
    poolFilter === "stable"
      ? stablePools
      : poolFilter === "volatile"
      ? volatilePools
      : poolFilter === "gohm"
      ? gOHMPools
      : defiLlamaPools;

  const poolListByNetwork = networkFilter ? poolList?.filter(pool => pool.chain === networkFilter) : poolList;
  const PoolChip = ({ label }: { label: string }) => (
    <Chip
      label={
        <Typography fontWeight="500" fontSize="12px">
          {poolFilter === label.toLowerCase() && <Check sx={{ fontSize: "12px", marginRight: "3px" }} />}
          {label}
        </Typography>
      }
      template={poolFilter === label.toLowerCase() ? undefined : "gray"}
      onClick={() => (poolFilter === label.toLowerCase() ? setPoolFilter("all") : setPoolFilter(label.toLowerCase()))}
    />
  );

  const columns: GridColDef<DefiLlamaPool>[] = [
    {
      field: "symbol",
      headerName: "Pool",
      renderCell: (params: GridRenderCellParams<DefiLlamaPool>) => {
        const symbols =
          params.row.symbol !== "OHMFRAXBP-F"
            ? params.row.symbol.split("-").filter(s => s !== "")
            : ["OHM", "FRAX", "CRV"];
        return (
          <StyledPoolInfo className={classes.poolPair}>
            <TokenStack
              tokens={normalizeSymbol(symbols) as OHMTokenStackProps["tokens"]}
              style={{ fontSize: "27px" }}
              network={defiLlamaChainToNetwork(params.row.chain) as OHMTokenProps["name"]}
            />

            <div className={classes.poolName}>
              <Typography fontWeight={700}>{params.row.symbol}</Typography>
            </div>
          </StyledPoolInfo>
        );
      },
      minWidth: 250,
      flex: 1,
    },
    {
      field: "tvlUsd",
      headerName: "TVL",
      valueFormatter: (params: GridValueFormatterParams) => formatCurrency(params.value, 0),
      minWidth: 200,
    },
    {
      field: "apy",
      headerName: "APY",
      renderCell: (params: GridRenderCellParams<DefiLlamaPool>) => (
        <>
          {params.row.apyBase || params.row.apyReward ? (
            <Tooltip
              message={
                <>
                  <p>Base APY: {formatNumber(params.row.apyBase || 0, 2)}%</p>
                  <p>Reward APY: {formatNumber(params.row.apyReward || 0, 2)}%</p>
                </>
              }
            >
              {formatNumber(params.row.apy || 0, 2)}%
            </Tooltip>
          ) : (
            <>{formatNumber(params.row.apy || 0, 2)}%</>
          )}
        </>
      ),
    },
    {
      field: "projectName",
      headerName: "Project Name",
      renderCell: (params: GridRenderCellParams<DefiLlamaPool>) => params.row.projectName,
      minWidth: 200,
    },
    {
      field: "id",
      headerName: "",
      renderCell: (params: GridRenderCellParams<DefiLlamaPool>) => (
        <TertiaryButton href={`https://defillama.com/yields/pool/${params.row.id}`} size="small">
          View Details
        </TertiaryButton>
      ),
      sortable: false,
      minWidth: 200,
    },
  ];

  return (
    <>
      <PageTitle
        name="Liquidity Pools"
        subtitle="Increase OHM's use in DeFi by pairing your OHM with other ERC-20 tokens and provide liquidity"
        noMargin
      />

      <Box display="flex" justifyContent="space-between" alignItems="center" flexWrap="wrap" gap={2}>
        <Box display="flex" gap="9px">
          <PoolChip label="All" />
          <PoolChip label="Stable" />
          <PoolChip label="Volatile" />
          <PoolChip label="gOHM" />
        </Box>
      </Box>
      <DataGrid
        columns={columns}
        rows={poolListByNetwork || []}
        autoHeight
        sx={{
          border: 0,
          fontSize: "15px",
          "& .MuiDataGrid-cell": {
            borderBottom: "none",
          },
          "& .MuiDataGrid-withBorderColor": {
            border: "none",
          },
          "& .MuiDataGrid-cell:focus-within": {
            outline: "none",
          },
          "& .MuiDataGrid-iconSeparator": {
            display: "none",
          },
          "& .MuiDataGrid-cell:focus, & .MuiDataGrid-columnHeader:focus, & .MuiDataGrid-columnHeader:focus-within": {
            outline: "none",
          },
        }}
        initialState={{
          sorting: {
            sortModel: [{ field: "tvlUsd", sort: "desc" }],
          },
          pagination: {
            paginationModel: {
              pageSize: 3,
            },
          },
        }}
        disableRowSelectionOnClick
        rowHeight={58}
        disableColumnMenu
        hideFooter
      />
      <Box display="flex" justifyContent="center">
        <SecondaryButton href="https://defillama.com/yields?token=GOHM&token=OHM">
          Explore More on DefiLlama
        </SecondaryButton>
      </Box>
    </>
  );
};
