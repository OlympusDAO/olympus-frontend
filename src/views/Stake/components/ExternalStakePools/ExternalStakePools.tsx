import { t, Trans } from "@lingui/macro";
import { Box, makeStyles, Table, TableCell, TableHead, TableRow, Typography, Zoom } from "@material-ui/core";
import useMediaQuery from "@material-ui/core/useMediaQuery";
import { Skeleton } from "@material-ui/lab";
import { DataRow, OHMTokenProps, Paper, SecondaryButton, Token, TokenStack } from "@olympusdao/component-library";
import { formatCurrency, formatNumber } from "src/helpers";
import { beetsPools, joePools, jonesPools, spiritPools, sushiPools, zipPools } from "src/helpers/AllExternalPools";
import { useWeb3Context } from "src/hooks/web3Context";
import { ExternalPool } from "src/lib/ExternalPool";
import { NetworkId } from "src/networkDetails";

import {
  BeetsPoolAPY,
  JoePoolAPY,
  JonesPoolAPY,
  SpiritPoolAPY,
  SushiPoolAPY,
  ZipPoolAPY,
} from "./hooks/useStakePoolAPY";
import { useStakePoolBalance } from "./hooks/useStakePoolBalance";
import { BalancerPoolTVL, useStakePoolTVL } from "./hooks/useStakePoolTVL";

const useStyles = makeStyles(theme => ({
  stakePoolsWrapper: {
    display: "grid",
    gridTemplateColumns: `1.0fr 0.5fr 0.5fr 1.5fr auto`,
    gridTemplateRows: "auto",
    alignItems: "center",
  },
  stakePoolHeaderText: {
    color: theme.palette.text.secondary,
    lineHeight: 1.4,
  },
  poolPair: {
    display: "flex !important",
    alignItems: "center",
    justifyContent: "left",
    marginBottom: "15px",
  },
  poolName: {
    marginLeft: "10px",
  },
}));

const AllPools = (props: { isSmallScreen: boolean }) => (
  <>
    {sushiPools.map(pool => (
      <SushiPools pool={pool} isSmallScreen={props.isSmallScreen} />
    ))}
    {joePools.map(pool => (
      <JoePools pool={pool} isSmallScreen={props.isSmallScreen} />
    ))}
    {spiritPools.map(pool => (
      <SpiritPools pool={pool} isSmallScreen={props.isSmallScreen} />
    ))}
    {beetsPools.map(pool => (
      <BeetsPools pool={pool} isSmallScreen={props.isSmallScreen} />
    ))}
    {zipPools.map(pool => (
      <ZipPools pool={pool} isSmallScreen={props.isSmallScreen} />
    ))}
    {jonesPools.map(pool => (
      <JonesPools pool={pool} isSmallScreen={props.isSmallScreen} />
    ))}
  </>
);

const ExternalStakePools = () => {
  const styles = useStyles();
  const { connected } = useWeb3Context();
  const isSmallScreen = useMediaQuery("(max-width: 705px)");

  return (
    <Zoom in={true}>
      {isSmallScreen ? (
        <AllPools isSmallScreen={isSmallScreen} />
      ) : (
        <Paper headerText={t`Farm Pool`}>
          <Table>
            <TableHead className={styles.stakePoolHeaderText}>
              <TableRow>
                <TableCell align="center">
                  <Trans>Asset</Trans>
                </TableCell>
                <TableCell>
                  <Trans>TVL</Trans>
                </TableCell>
                <TableCell>
                  <Trans>APY</Trans>
                </TableCell>
                <TableCell>{connected ? t`Balance` : ""}</TableCell>
              </TableRow>
            </TableHead>
            <AllPools isSmallScreen={isSmallScreen} />
          </Table>
        </Paper>
      )}
    </Zoom>
  );
};

const StakePool: React.FC<{ pool: ExternalPool; tvl?: number; apy?: number }> = props => {
  const { connected } = useWeb3Context();

  const userBalances = useStakePoolBalance(props.pool);
  const userBalance = userBalances[props.pool.networkID].data;

  return (
    <TableRow>
      <TableCell>
        <Box display="flex" flexDirection="row" alignItems="center">
          <TokenStack tokens={props.pool.icons} />
          <Typography gutterBottom={false} style={{ lineHeight: 1.4, marginLeft: "10px", marginRight: "10px" }}>
            {props.pool.poolName}
          </Typography>
          <Token name={NetworkId[props.pool.networkID] as OHMTokenProps["name"]} style={{ fontSize: "15px" }} />
        </Box>
      </TableCell>
      <TableCell>
        <Typography gutterBottom={false} style={{ lineHeight: 1.4 }}>
          {!props.tvl ? <Skeleton width={80} /> : formatCurrency(props.tvl)}
        </Typography>
      </TableCell>
      <TableCell>
        <Typography gutterBottom={false} style={{ lineHeight: 1.4 }}>
          {!props.apy ? 0 : formatNumber(props.apy * 100, 2)}%
        </Typography>
      </TableCell>
      <TableCell>
        <Typography gutterBottom={false} style={{ lineHeight: 1.4 }}>
          {!connected ? (
            ""
          ) : !userBalance ? (
            <Skeleton width={80} />
          ) : (
            `${userBalance.toString({ decimals: 4, trim: false, format: true })} LP`
          )}
        </Typography>
      </TableCell>
      <TableCell>
        <SecondaryButton size="small" target="_blank" href={props.pool.href} fullWidth>
          {t`Stake on`} {props.pool.stakeOn}
        </SecondaryButton>
      </TableCell>
    </TableRow>
  );
};

const MobileStakePool: React.FC<{ pool: ExternalPool; tvl?: number; apy?: number }> = props => {
  const styles = useStyles();
  const { connected } = useWeb3Context();

  const userBalances = useStakePoolBalance(props.pool);
  const userBalance = userBalances[props.pool.networkID].data;

  return (
    <Paper>
      <div className={styles.poolPair}>
        <TokenStack tokens={props.pool.icons} />

        <div className={styles.poolName}>
          <Typography>{props.pool.poolName}</Typography>
        </div>
        <div className={styles.poolName}>
          <Token name={NetworkId[props.pool.networkID] as OHMTokenProps["name"]} style={{ fontSize: "15px" }} />
        </div>
      </div>

      <DataRow title={`TVL`} isLoading={!props.tvl} balance={props.tvl ? formatCurrency(props.tvl) : undefined} />
      <DataRow
        title={`APY`}
        isLoading={!props.apy}
        balance={props.apy ? `${formatNumber(props.apy * 100, 2)} %` : "0"}
      />

      {connected && (
        <DataRow
          title={t`Balance`}
          isLoading={!userBalance}
          balance={userBalance && `${userBalance.toString({ decimals: 4, trim: false, format: true })} LP`}
        />
      )}

      <SecondaryButton href={props.pool.href} fullWidth>
        {t`Stake on`} {props.pool.stakeOn}
      </SecondaryButton>
    </Paper>
  );
};

const SushiPools: React.FC<{ pool: ExternalPool; isSmallScreen: boolean }> = props => {
  const { data: totalValueLocked } = useStakePoolTVL(props.pool);
  const { apy } = SushiPoolAPY(props.pool);
  return props.isSmallScreen ? (
    <MobileStakePool pool={props.pool} tvl={totalValueLocked} apy={apy} />
  ) : (
    <StakePool pool={props.pool} tvl={totalValueLocked} apy={apy} />
  );
};

const JoePools: React.FC<{ pool: ExternalPool; isSmallScreen: boolean }> = props => {
  const { data: totalValueLocked } = useStakePoolTVL(props.pool);
  const { apy } = JoePoolAPY(props.pool);
  return props.isSmallScreen ? (
    <MobileStakePool pool={props.pool} tvl={totalValueLocked} apy={apy} />
  ) : (
    <StakePool pool={props.pool} tvl={totalValueLocked} apy={apy} />
  );
};
const SpiritPools: React.FC<{ pool: ExternalPool; isSmallScreen: boolean }> = props => {
  const { data: totalValueLocked } = useStakePoolTVL(props.pool);
  const { apy } = SpiritPoolAPY(props.pool);
  return props.isSmallScreen ? (
    <MobileStakePool pool={props.pool} tvl={totalValueLocked} apy={apy} />
  ) : (
    <StakePool pool={props.pool} tvl={totalValueLocked} apy={apy} />
  );
};
const BeetsPools: React.FC<{ pool: ExternalPool; isSmallScreen: boolean }> = props => {
  const { data: totalValueLocked } = BalancerPoolTVL(props.pool);
  const { apy } = BeetsPoolAPY(props.pool);
  return props.isSmallScreen ? (
    <MobileStakePool pool={props.pool} tvl={totalValueLocked} apy={apy} />
  ) : (
    <StakePool pool={props.pool} tvl={totalValueLocked} apy={apy} />
  );
};

const ZipPools: React.FC<{ pool: ExternalPool; isSmallScreen: boolean }> = props => {
  const { data: totalValueLocked } = useStakePoolTVL(props.pool);
  const { apy } = ZipPoolAPY(props.pool);
  return props.isSmallScreen ? (
    <MobileStakePool pool={props.pool} tvl={totalValueLocked} apy={apy} />
  ) : (
    <StakePool pool={props.pool} tvl={totalValueLocked} apy={apy} />
  );
};

const JonesPools: React.FC<{ pool: ExternalPool; isSmallScreen: boolean }> = props => {
  const { data: totalValueLocked } = useStakePoolTVL(props.pool);
  const { apy } = JonesPoolAPY(props.pool);
  return props.isSmallScreen ? (
    <MobileStakePool pool={props.pool} tvl={totalValueLocked} apy={apy} />
  ) : (
    <StakePool pool={props.pool} tvl={totalValueLocked} apy={apy} />
  );
};

export default ExternalStakePools;
