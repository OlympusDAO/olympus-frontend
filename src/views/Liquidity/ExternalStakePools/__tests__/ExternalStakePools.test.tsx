import { render, screen } from "src/testUtils";
import { ExternalStakePools } from "src/views/Liquidity/ExternalStakePools/ExternalStakePools";
import { describe, expect, it } from "vitest";

describe("ExternalStakePools", () => {
  it("renders all pool chips", () => {
    render(<ExternalStakePools />);

    const allPoolChip = screen.getByText("All");
    const stablePoolChip = screen.getByText("Stable");
    const volatilePoolChip = screen.getByText("Volatile");
    const gohmPoolChip = screen.getByText("gOHM");

    expect(allPoolChip).not.toBeNull();
    expect(stablePoolChip).not.toBeNull();
    expect(volatilePoolChip).not.toBeNull();
    expect(gohmPoolChip).not.toBeNull();
  });
});
