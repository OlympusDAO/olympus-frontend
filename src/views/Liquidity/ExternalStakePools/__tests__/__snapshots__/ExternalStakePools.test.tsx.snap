// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ExternalStakePools > renders snapshot correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="position: fixed; z-index: 9999; top: 16px; left: 16px; right: 16px; bottom: 16px; pointer-events: none;"
      />
      <div
        id="stake-view"
      >
        <div
          class="MuiBox-root css-6y4wdt"
        >
          <h1
            class="MuiTypography-root MuiTypography-h1 css-f2reur-MuiTypography-root"
          >
            <div
              class="MuiBox-root css-u4p24i"
            >
              <div
                class="MuiBox-root css-1eek0hz"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1xtlg2o-MuiTypography-root"
                >
                  Liquidity Pools
                </p>
              </div>
            </div>
          </h1>
        </div>
        <div
          class="MuiBox-root css-qmy2sf"
        >
          <div
            class="MuiBox-root css-j3ty86"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-pi8uen-MuiTypography-root"
            >
              Increase OHM's use in DeFi by pairing your OHM with other ERC-20 tokens and provide liquidity
            </p>
          </div>
          <div
            class="MuiBox-root css-nwtebv"
          >
            <div
              class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-filledDefault Chip-chip css-1oagpq9-MuiButtonBase-root-MuiChip-root"
              role="button"
              tabindex="0"
            >
              <span
                class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-vrumy4-MuiTypography-root"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-l0e0t4-MuiSvgIcon-root"
                    data-testid="CheckIcon"
                    focusable="false"
                    viewBox="0 0 20 20"
                  >
                    <path
                      d="M9 16.17 4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
                    />
                  </svg>
                  All
                </p>
              </span>
            </div>
            <div
              class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-filledDefault Chip-chip css-11c5anp-MuiButtonBase-root-MuiChip-root"
              role="button"
              tabindex="0"
            >
              <span
                class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-vrumy4-MuiTypography-root"
                >
                  Stable
                </p>
              </span>
            </div>
            <div
              class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-filledDefault Chip-chip css-11c5anp-MuiButtonBase-root-MuiChip-root"
              role="button"
              tabindex="0"
            >
              <span
                class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-vrumy4-MuiTypography-root"
                >
                  Volatile
                </p>
              </span>
            </div>
            <div
              class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-filledDefault Chip-chip css-11c5anp-MuiButtonBase-root-MuiChip-root"
              role="button"
              tabindex="0"
            >
              <span
                class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-vrumy4-MuiTypography-root"
                >
                  gOHM
                </p>
              </span>
            </div>
          </div>
          <table
            class="MuiTable-root css-dcenn1-MuiTable-root"
          >
            <thead
              class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
            >
              <tr
                class="MuiTableRow-root MuiTableRow-head css-xjst8z-MuiTableRow-root"
              >
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-a3zox6-MuiTableCell-root"
                  scope="col"
                >
                  Asset
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-a3zox6-MuiTableCell-root"
                  scope="col"
                >
                  TVL
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-a3zox6-MuiTableCell-root"
                  scope="col"
                >
                  APY
                  <div
                    class="MuiBox-root css-0"
                    style="display: inline-flex; justify-content: center; align-self: center;"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall info-icon css-1b8e8sl-MuiSvgIcon-root"
                      focusable="false"
                      style="margin: 0px 5px; font-size: 1em;"
                      viewBox="0 0 20 20"
                    >
                      <path
                        d="M 10 20 C 15.475 20 20 15.473 20 10 C 20 4.518 15.473 0 9.991 0 C 4.516 0 0 4.518 0 10 C 0 15.475 4.527 20 10 20 Z M 10 18.705 C 5.189 18.714 1.287 14.812 1.297 10.001 C 1.28 5.189 5.179 1.281 9.991 1.285 C 14.807 1.28 18.714 5.182 18.714 9.999 C 18.719 14.811 14.813 18.712 10 18.702 Z M 9.941 6.242 C 10.559 6.242 11.038 5.763 11.038 5.156 C 11.043 4.547 10.549 4.053 9.94 4.058 C 9.334 4.057 8.842 4.55 8.844 5.156 C 8.843 5.761 9.337 6.249 9.941 6.242 Z M 8.216 15.444 L 12.303 15.444 C 12.623 15.444 12.871 15.204 12.871 14.895 C 12.87 14.584 12.615 14.334 12.303 14.338 L 10.868 14.338 L 10.868 8.754 C 10.868 8.346 10.658 8.065 10.269 8.065 L 8.345 8.065 C 7.919 8.045 7.631 8.493 7.826 8.872 C 7.925 9.065 8.128 9.182 8.345 9.172 L 9.652 9.172 L 9.652 14.338 L 8.216 14.338 C 7.905 14.334 7.649 14.584 7.648 14.895 C 7.648 15.204 7.897 15.444 8.216 15.444 Z"
                      />
                    </svg>
                  </div>
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-ngk6ek-MuiTableCell-root"
                  scope="col"
                  width="100px"
                />
              </tr>
            </thead>
            <tbody
              class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
            />
          </table>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="position: fixed; z-index: 9999; top: 16px; left: 16px; right: 16px; bottom: 16px; pointer-events: none;"
    />
    <div
      id="stake-view"
    >
      <div
        class="MuiBox-root css-6y4wdt"
      >
        <h1
          class="MuiTypography-root MuiTypography-h1 css-f2reur-MuiTypography-root"
        >
          <div
            class="MuiBox-root css-u4p24i"
          >
            <div
              class="MuiBox-root css-1eek0hz"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-1xtlg2o-MuiTypography-root"
              >
                Liquidity Pools
              </p>
            </div>
          </div>
        </h1>
      </div>
      <div
        class="MuiBox-root css-qmy2sf"
      >
        <div
          class="MuiBox-root css-j3ty86"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-pi8uen-MuiTypography-root"
          >
            Increase OHM's use in DeFi by pairing your OHM with other ERC-20 tokens and provide liquidity
          </p>
        </div>
        <div
          class="MuiBox-root css-nwtebv"
        >
          <div
            class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-filledDefault Chip-chip css-1oagpq9-MuiButtonBase-root-MuiChip-root"
            role="button"
            tabindex="0"
          >
            <span
              class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-vrumy4-MuiTypography-root"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-l0e0t4-MuiSvgIcon-root"
                  data-testid="CheckIcon"
                  focusable="false"
                  viewBox="0 0 20 20"
                >
                  <path
                    d="M9 16.17 4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
                  />
                </svg>
                All
              </p>
            </span>
          </div>
          <div
            class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-filledDefault Chip-chip css-11c5anp-MuiButtonBase-root-MuiChip-root"
            role="button"
            tabindex="0"
          >
            <span
              class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-vrumy4-MuiTypography-root"
              >
                Stable
              </p>
            </span>
          </div>
          <div
            class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-filledDefault Chip-chip css-11c5anp-MuiButtonBase-root-MuiChip-root"
            role="button"
            tabindex="0"
          >
            <span
              class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-vrumy4-MuiTypography-root"
              >
                Volatile
              </p>
            </span>
          </div>
          <div
            class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-filledDefault Chip-chip css-11c5anp-MuiButtonBase-root-MuiChip-root"
            role="button"
            tabindex="0"
          >
            <span
              class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-vrumy4-MuiTypography-root"
              >
                gOHM
              </p>
            </span>
          </div>
        </div>
        <table
          class="MuiTable-root css-dcenn1-MuiTable-root"
        >
          <thead
            class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head css-xjst8z-MuiTableRow-root"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-a3zox6-MuiTableCell-root"
                scope="col"
              >
                Asset
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-a3zox6-MuiTableCell-root"
                scope="col"
              >
                TVL
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-a3zox6-MuiTableCell-root"
                scope="col"
              >
                APY
                <div
                  class="MuiBox-root css-0"
                  style="display: inline-flex; justify-content: center; align-self: center;"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall info-icon css-1b8e8sl-MuiSvgIcon-root"
                    focusable="false"
                    style="margin: 0px 5px; font-size: 1em;"
                    viewBox="0 0 20 20"
                  >
                    <path
                      d="M 10 20 C 15.475 20 20 15.473 20 10 C 20 4.518 15.473 0 9.991 0 C 4.516 0 0 4.518 0 10 C 0 15.475 4.527 20 10 20 Z M 10 18.705 C 5.189 18.714 1.287 14.812 1.297 10.001 C 1.28 5.189 5.179 1.281 9.991 1.285 C 14.807 1.28 18.714 5.182 18.714 9.999 C 18.719 14.811 14.813 18.712 10 18.702 Z M 9.941 6.242 C 10.559 6.242 11.038 5.763 11.038 5.156 C 11.043 4.547 10.549 4.053 9.94 4.058 C 9.334 4.057 8.842 4.55 8.844 5.156 C 8.843 5.761 9.337 6.249 9.941 6.242 Z M 8.216 15.444 L 12.303 15.444 C 12.623 15.444 12.871 15.204 12.871 14.895 C 12.87 14.584 12.615 14.334 12.303 14.338 L 10.868 14.338 L 10.868 8.754 C 10.868 8.346 10.658 8.065 10.269 8.065 L 8.345 8.065 C 7.919 8.045 7.631 8.493 7.826 8.872 C 7.925 9.065 8.128 9.182 8.345 9.172 L 9.652 9.172 L 9.652 14.338 L 8.216 14.338 C 7.905 14.334 7.649 14.584 7.648 14.895 C 7.648 15.204 7.897 15.444 8.216 15.444 Z"
                    />
                  </svg>
                </div>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-ngk6ek-MuiTableCell-root"
                scope="col"
                width="100px"
              />
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
          />
        </table>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
